Controller.cpp

```diff
@@ -0,0 +1,264 @@
+
+#include "Controller.h"
+
+#include <QCoreApplication>
+#include <QKeyEvent>
+#include <QTime>
+
+
+
+Controller::Controller(Robot::Model robotModel, Maze::Model mazeModel, QObject *parent) :
+    QObject(parent), m_robotModel(robotModel), m_mazeModel(mazeModel)
+{
+    if(!m_states.isEmpty()) { m_states.clear();}
+    m_states.push(State(m_robotModel.robotPosition,
```

Comment:
мне кажется можно прям 
```suggestion
    m_states.push(m_robotModel); 
```
там все равно копирование будет вызвано
у тебя же QStack не ссылки хранит, вообще судя по коду есть непонимание про вот эти все конструкторы копирования перемещения что происходит при вызове функций и тп
хз даже что почитать
 мб это расширит понимание
 https://habr.com/ru/post/646005/